packages/Python/lldbsuite/test/lang/swift/implementation_only_imports/main_executable/TestMainExecutable.py

```diff
@@ -0,0 +1,86 @@
+import os
+import os.path
+import time
+import unittest2
+
+class TestMainExecutable(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    def launch_and_break(self):
+        exe_name = "main"
+        exe_path = self.getBuildArtifact(exe_name)
+        target = self.dbg.CreateTarget(exe_path)
+        self.assertTrue(target, VALID_TARGET)
+        
+        source_name = "main.swift"
+        source_spec = lldb.SBFileSpec(source_name)
+        breakpoint = target.BreakpointCreateBySourceRegex("break here", source_spec)
+        self.assertTrue(breakpoint.GetNumLocations() > 0, "Breakpoint set sucessfully")
+        
+        process = target.LaunchSimple(None, None, os.getcwd())
+        self.assertTrue(process, PROCESS_IS_VALID)
+    
+    @decorators.swiftTest
+    def test_implementation_only_import_main_executable(self):
+        """Test `@_implementationOnly import` in the main executable"""
+        self.build()
+        def cleanup():
+            lldbutil.execute_command("make cleanup")
+        self.addTearDownHook(cleanup)
+        self.launch_and_break()
+
+        self.expect("fr var", substrs=[
+            "(SomeLibrary.TwoInts) value = (first = 2, second = 3)",
+            "(main.ContainsTwoInts) container = {\n  wrapped = (first = 2, second = 3)\n  other = 10\n}"])
+        self.expect("e value", substrs=["(SomeLibrary.TwoInts)", "= (first = 2, second = 3)"])
+        self.expect("e container", substrs=["(main.ContainsTwoInts)", "wrapped = (first = 2, second = 3)", "other = 10"])
+        self.expect("e TwoInts(4, 5)", substrs=["(SomeLibrary.TwoInts)", "= (first = 4, second = 5)"])
+    
+    @decorators.swiftTest
+    def test_implementation_only_import_main_executable_no_library_module(self):
+        """Test `@_implementationOnly import` in the main executable, after removing the library's swiftmodule"""
+        self.build()
+        os.remove(os.path.join(self.getBuildDir(), "SomeLibrary.swiftmodule"))
+        os.remove(os.path.join(self.getBuildDir(), "SomeLibrary.swiftinterface"))
+        def cleanup():
+            lldbutil.execute_command("make cleanup")
+        self.addTearDownHook(cleanup)
+        self.launch_and_break()
+
+        self.expect("fr var", substrs=[
+            "value = <could not resolve type>",
+        # FIXME: This representation of 'container' is garbage because we've dropped a field.
+        # The compiler should keep track of this.
+            "container = (other = 2)"])
+        # FIXME: Ideally we would still show that 'value' exists but make it unusable somehow.
+        self.expect("e value", error=True, substrs=["error: use of unresolved identifier 'value'"])
+        # FIXME: Same problem as above.
+        self.expect("e container", substrs=["(main.ContainsTwoInts)", "other = 2"])
+        # This one's probably unavoidable, though.
+        self.expect("e TwoInts(4, 5)", error=True, substrs=["error: use of unresolved identifier 'TwoInts'"])
+
+if __name__ == '__main__':
```

Comment:
You don't really need this `if __name__ == main` and below. Some tests have it, but it's just an historical artifact.